src/components/webPlayback.jsx

```diff
@@ -0,0 +1,156 @@
+import React from 'react';
+
+const track = {
+  name: "",
+  album: {
+    images: [
+      { url: "" }
+    ]
+  },
+  artists: [
+    { name: "" }
+  ]
+};
+
+
+class WebPlayback extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      is_paused: false,
+      is_active: false,
+      player: undefined,
+      is_playing_left_track: true,
+      left_side_track: track,
+      right_side_track: track,
+      previous_track: false
+    }
+
+    this.handlePrevSong = this.handlePrevSong.bind(this);
+    this.handleNextSong = this.handleNextSong.bind(this);
+  }
+
+  componentDidMount() {
+    const script = document.createElement("script");
+    script.src = 'https://sdk.scdn.co/spotify-player.js';
+    script.async = true;
+    document.body.appendChild(script);
+
+    window.onSpotifyWebPlaybackSDKReady = () => {
+      const player = new window.Spotify.Player({
+        name: 'Web Playback SDK',
+        getOAuthToken: cb => cb(this.props.token),
+        volume: 0.5
+      });
+
+      this.setState({ player: player });
+      
+      this.state.player.addListener('ready', ({ device_id }) => {
+        console.log('Ready with Device ID', device_id);
+      });
+
+      this.state.player.addListener('not_ready', ({ device_id }) => {
+        console.log('Device ID has gone offline', device_id);
+      });
+
+      this.state.player.addListener('player_state_changed', ( state => {
+        if (!state) {
+          return;
+        }
+
+        if (this.state.is_playing_left_track) {
+          this.setState({ left_side_track: state.track_window.current_track, right_side_track: state.track_window.next_tracks[0], is_paused: state.paused });
+        } else {
+          this.setState({ left_side_track: state.track_window.previous_tracks[1], right_side_track: state.track_window.current_track, is_paused: state.paused });
+        }
```

Comment:
you can tighten this up a bit by doing some [destructuring](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment) and using ternaries:

```javascript
const {is_playing_left_track, paused, track_window: {current_track, next_tracks, previous_tracks}} = state;
this.setState({left_side_track: is_playing_left_track ? current_track : previous_tracks[1], right_side_track: is_playing_left_track ? next_tracks[0] : current_track, is_paused:  paused});
```

or if you want it to be more readable across multiple lines:

```javascript
const {
  is_playing_left_track,
  paused,
  track_window: {current_track, next_tracks, previous_tracks}
} = state;

this.setState({
  left_side_track: is_playing_left_track ? current_track : previous_tracks[1],
  right_side_track: is_playing_left_track ? next_tracks[0] : current_track,
  is_paused:  paused
});

```